code/modules/projectiles/projectile/special/crab.dm

```diff
@@ -0,0 +1,121 @@
+#define HAT -1
+#define NOTNOTLIVE 0
+#define NOTNOTDEAD 1
+#define HEADCRAB_ZOMBIE_DEFAULT_MESSAGE "You feel a numbness across your new host's body. Your only instinct now is to kill the rest of this place's inhabitants"
+/obj/projectile/crab
+	name = "Headcrab"
+	icon_state = "crabby_weeee"
+	damage = 80
+	damage_type = STAMINA
+	def_zone = BODY_ZONE_HEAD
+	hitsound = 'sound/weapons/tap.ogg'
+	var/takenHealth = 0
+
+/obj/projectile/crab/proc/weHit(atom/hit)
+	//Initiate facehugger COPYPASTA
+	//probiscis-blocker handling
+	if(iscarbon(hit))
+		var/mob/living/carbon/victim = hit
+		if(victim.head && istype(victim.head, /obj/item/clothing/head/headcrab))
+			return NOTNOTLIVE
+		if(ishuman(hit))
+			var/mob/living/carbon/human/H = hit
+			if(H.is_mouth_covered(head_only = 1))
+				H.visible_message("<span class='danger'>[src] smashes against [H]'s [H.head]!</span>", \
+									"<span class='userdanger'>[src] smashes against your [H.head]!</span>", \
+									"<span class='userdanger'>You feel a hard object knock your head back!</span>")
+				return NOTNOTDEAD
+		if(victim.head)
+			var/obj/item/clothing/W = victim.head
+			if(victim.dropItemToGround(W))
+				victim.visible_message("<span class='danger'>[src] knocks [W] off of [victim]'s head!</span>", \
+									"<span class='userdanger'>[src] knocks [W] off of your head!</span>")
+		var/obj/item/clothing/head/headcrab/creb = new /obj/item/clothing/head/headcrab
+		obj_integrity = obj_integrity - (obj_integrity-takenHealth)
+		victim.equip_to_slot_if_possible(creb, ITEM_SLOT_HEAD, 0, 1, 1)
+		return HAT
+	return NOTNOTLIVE
+
+/obj/projectile/crab/proc/shed(kill)
+	var/mob/living/simple_animal/hostile/headcrustation/creb = new(get_turf(src))
+	creb.adjustHealth(creb.bruteloss - (creb.bruteloss-takenHealth))
+	if(kill)
+		creb.death(FALSE)
```

Comment:
These two procs can cause `death` to be called twice.

Reply:
That shouldn't be possible, unless the crab had negative damage on the initial shoot. The math should be correct, not totally sure on that tho